lib/application/widgets/theme/tags_field.dart

```diff
@@ -0,0 +1,273 @@
+        fieldController.removeListener(onSearchUpdated);
+        fieldFocus.removeListener(onFocusUpdate);
+        controller.removeListener(onTagsUpdate);
+      };
+    });
+
+    // Simulates `TextField.helperText` animating the label when the field is focused or not.
+    final helperTitle = AnimatedDefaultTextStyle(
+      style: isFocused.value ? textTheme.caption!.copyWith(color: theme.neutralSwatch.shade300) : textTheme.subtitle1!,
+      duration: anims.textFieldHelperTextDuration,
+      child: const Text(strings.addTags),
+    );
+
+    final tagsFieldContainer = Container(
+      constraints: dimens.richTextFieldConstraints,
+      decoration: BoxDecoration(
+        borderRadius: dimens.genericRoundedElementBorderRadius,
+        color: textInputDecoration.fillColor,
+        border: Border.all(
+          color: errorText != null ? theme.destructiveSwatch : textInputDecoration.fillColor!,
+          width: dimens.genericBorderHeight,
+        ),
+      ),
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          helperTitle,
+          if (isFocused.value) ...[
+            context.verticalBox(Spacing.small),
+            _TagsTextField(
+              controller: controller,
+              fieldController: fieldController,
+              focus: fieldFocus,
+              onSubmitted: onTagSelected,
+            ),
+          ]
+        ],
+      ).withSymmetricalPadding(context, vertical: Spacing.small, horizontal: Spacing.medium),
+    );
+
+    final helperText = Text(
+      errorText != null ? errorText! : strings.tagsAmount(tags.value.length, maxTags),
+      style: textTheme.caption?.copyWith(
+        color: errorText != null ? theme.destructiveSwatch : theme.neutralSwatch.shade400,
+      ),
+    );
+
+    return GestureDetector(
+      onTap: () {
+        fieldFocus.requestFocus();
+        isFocused.value = true;
+      },
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          tagsFieldContainer,
+          context.verticalBox(Spacing.xxxSmall),
+          helperText.withOnlyPadding(context, left: Spacing.small)
+        ],
+      ),
+    );
+  }
+}
+
+/// The custom tagged TextField.
+///
+/// Wraps the already chosen tags with an input [TextField] used to add new tags.
+class _TagsTextField extends HookConsumerWidget {
+  const _TagsTextField({required this.controller, this.fieldController, this.focus, this.onSubmitted});
+
+  final TagsController controller;
+  final TextEditingController? fieldController;
+
+  /// Controls the current field focus.
+  final FocusNode? focus;
+
+  /// Called when the user indicates they are done inputting `tag`.
+  ///
+  /// Usually fired when the user press the "done" keyboard button.
+  final Function(String tag)? onSubmitted;
```

Comment:
```suggestion
  final void Function(String tag)? onSubmitted;
```

Reply:
Can't we have a lint for that? I'll take a look